runtime/interpreter/function.go

```diff
@@ -61,6 +62,8 @@ type InterpretedFunctionValue struct {
 	PreConditions    ast.Conditions
 	Statements       []ast.Statement
 	PostConditions   ast.Conditions
+	staticType       StaticType
+	staticTypeOnce   sync.Once
```

Comment:
There is no concurrency in the interpreter and the interpreter is not used in a concurrent way, so using sync.Once is not necessary (here and for the other cases below)

Reply:
There are some functions that we use across interpreters (e.g: [typeFunction](https://github.com/onflow/cadence/blob/master/runtime/interpreter/interpreter.go#L2638-L2640), etc ) could that cause issues?

Reply:
Ah, that's a good point, I didn't think of that! I guess you are right, which is unfortunate. I guess we should maybe refactor things like that to be independent for each interpreter, instead globally shared, potentially even used concurrently